ibc-testkit/src/testapp/ibc/core/types.rs

```diff
@@ -469,39 +469,31 @@ where
     HostClientState<H>: ClientStateValidation<DefaultIbcStore>,
 {
     pub fn with_latest_height(height: Height) -> Self {
-        let context = TestContextConfig::builder()
+        let context: TestContext<_> = dummy_store_generic_test_context()
             .latest_height(height)
-            .build::<TestContext<_>>();
+            .build();
         LightClientBuilder::init().context(&context).build()
     }
 }
 
-#[derive(TypedBuilder)]
-#[builder(builder_method(name = init), build_method(into))]
-pub struct LightClientBuilder<'a, H>
-where
-    H: TestHost,
-    HostClientState<H>: ClientStateValidation<DefaultIbcStore>,
-{
-    context: &'a TestContext<H>,
-    #[builder(default, setter(into))]
-    consensus_heights: Vec<Height>,
-    #[builder(default)]
-    params: H::LightClientParams,
-}
+pub struct LightClientBuilder;
 
-impl<'a, H> From<LightClientBuilder<'a, H>> for LightClientState<H>
-where
-    H: TestHost,
-    HostClientState<H>: ClientStateValidation<DefaultIbcStore>,
-{
-    fn from(builder: LightClientBuilder<'a, H>) -> Self {
-        let LightClientBuilder {
-            context,
-            consensus_heights,
-            params,
-        } = builder;
+#[bon]
+impl LightClientBuilder {
+    #[builder(finish_fn = build)]
+    pub fn init<H>(
+        context: &TestContext<H>,
+
+        #[builder(default, into)] //
+        consensus_heights: Vec<Height>,
 
+        #[builder(default)] //
+        params: H::LightClientParams,
+    ) -> LightClientState<H>
+    where
+        H: TestHost,
+        HostClientState<H>: ClientStateValidation<DefaultIbcStore>,
+    {
```

Comment:
hey. can you explain why you still needed a struct here ? not `dummy_light_client_builder` method ?

Reply:
I think, I just hesitated here a bit, since it seemed to me like it wasn't just a dummy fixture. Now that you pointed this out, I'll change it to `dummy_light_client()`.

Reply:
Btw. I made `context` here to be the parameter of the starting function, so the syntax is:

```rust
dummy_light_client(context) // context is passed to the starting function
     .consensus_heights(...) // optional
     .params(...) // optional
     .build()
```

Reply:
Ah nice ! I can think of some places, we could do this. But I will take care of it. :raised_hands: 

Also, `build` is the default name for finish function, right ? In that case, we can remove the `(finish_fn = build)`.

Reply:
`build` is the default name for finish function if you derive the builder for a struct with `derive(bon::Builder)` or if you derive the builder from the associated method named `new`.

For other functions and methods that can execute any other arbitrary logic the default finishing method name is `call()`, because `bon` can't assume that the general-purpose functions/method the builder is generated on just do simple object building. In some cases like in [`ballistics_rs`](https://docs.rs/ballistics_rs/latest/ballistics_rs/) crate the functions do some pure calculations and return primitive values, in other cases people use `bon` on associated methods of async API clients `client.method().param().call().await`.

Reply:
Sure, I guess all things are resolved, and this is ready to merge then